corehq/util/celery_utils.py

```diff
@@ -186,3 +190,41 @@ def get_running_workers(timeout=10):
         worker_names.extend(worker_info.keys())
 
     return worker_names
+
+
+class LoadBasedAutoscaler(Autoscaler):
+    def _maybe_scale(self, req=None):
+        procs = self.processes
+        cur = min(self.qty, self.max_concurrency)
+
+        available_cpus = multiprocessing.cpu_count()
+        load_avgs = os.getloadavg()
+        max_avg = max(load_avgs[0], load_avgs[1])
+        normalized_load = max_avg / available_cpus
+
+        if cur > procs and normalized_load < 0.90:
+            self.scale_up(cur - procs)
+            return True
+        elif cur < procs or normalized_load > 0.90:
+            self.scale_down((procs - cur) - self.min_concurrency)
+            return True
+
+
+class OffPeakLoadBasedAutoscaler(LoadBasedAutoscaler):
+    def _is_off_peak(self):
+        now = datetime.utcnow().time()
+        if settings.OFF_PEAK_TIME:
+            if settings.OFF_PEAK_TIME[0] < now < settings.OFF_PEAK_TIME[1]:
+                return True
+
+        return False
+
+    def _maybe_scale(self, req=None):
+        procs = self.processes
+        cur = min(self.qty, self.max_concurrency)
+
+        if not self._is_off_peak():
+            self.scale_down((procs - cur) - self.min_concurrency)
+            return True
```

Comment:
is the intention that we'd scale down to min_concurrency when we're not off peak? if so then i think this should scale down procs - self.min_concurrency, but also should check that procs > self.min_concurrency before doing so